mautic/mautic
# Fix campaign scheduling hours range during change daylight saving time

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



#### Steps to test this PR:
In Europe two times per year is DST when hours are changed https://www.timeanddate.com/news/time/europe-dst-end-2021.html
We noticed the change this year on october 31. october made weird some scheduled events with hours range.  


1. Set Europer/Paris in timezone in Configuration
2. Create basic campaign with decision and action on negative path  for example
![image](https://user-images.githubusercontent.com/462477/141848769-e224f53d-5eaf-470b-846e-0e21975164b0.png)
3. Schedule negative patch for hours from 7 to 22 for example
4. Run campaign
5. Go to database and set date_triggered on the scheduled action before 31. october and set scheduled trigger_date for example on 7:30 day after 31. october
6.  You have to test it after 7:30 
7.  You see like your action was not trigger, just rescheduled
8. After PR should work






<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10612"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
